packages/nimble-components/src/dialog/styles.ts

```diff
@@ -39,6 +39,11 @@ export const styles = css`
         display: flex;
     }
 
+    :host([large-mode]) dialog {
```

Comment:
It's worth having a quick discussion about whether the `large-mode` attribute is the right API for this. We don't have great consensus on sizing APIs within Nimble yet. 
1. some components like the spinner let you set the `size` style directly
2. some components like the drawer expose a CSS variable that you can set
3. some components expose an enum attribute that lets you set different pre-set configurations (I can't recall any for size but we have examples of this pattern for `appearance` on many components, for example

I'm sure @rajsite will have opinions between these (and other) options. My own first thought is that an attribute like `size-mode` with values of `undefined` (for the current default behavior) and `'large'` (for the new behavior) would be more extendable if we decide to add more sizes in the future.

Reply:
So, would an approach similar to the way the `nimble-drawer` sets its width value be more suitable? However, as I understood having too many options for the dialog's size isn't something we want to support.

Reply:
I spoke to @rajsite about this offline and we decided it would make sense to take the API in a different direction. We still don't know what use cases we want the dialog to support out of the box and we're reluctant to expose a pre-set size like this until we know that more than one app would use it.

Until we have more use cases, our preferred approach would be to allow clients to set the width and height of the dialog from their code, similar to option 1 above and similar to what the spinner does already. I captured more details in #1117.

@CuckooHashing is your app development blocked on this feature or do you have a workaround already? Are you interested in refactoring this PR with this new recommended approach or would you rather we leave it in the Nimble backlog to be tackled at some future date?

Sorry for the delay in deciding this and for pivoting you away from your initial approach.